src/scheduler/pbs_sched_utils.cpp

```diff
@@ -574,11 +574,12 @@ connect_svrpool()
 			continue;
 		}
 
-		for (i = 0; i < num_conf_svrs; i++) {
-			if (svr_conns_primary[i].state == SVR_CONN_STATE_DOWN || svr_conns_primary[i].state == SVR_CONN_STATE_DOWN) {
-				if (svr_conns_primary[i].state == SVR_CONN_STATE_DOWN)
+		for (i = 0; svr_conns_primary[i] && svr_conns_secondary[i]; i++) {
+			if (svr_conns_primary[i]->state == SVR_CONN_STATE_DOWN ||
+			    svr_conns_primary[i]->state == SVR_CONN_STATE_DOWN) {
```

Comment:
This is how it appears in the mainline. I am removing this outer check as it is also handled in the inner if.

Reply:
Looks to be a bug in mainline then.  I suspect the if is supposed to check if the primary state is down or the secondary state is down.  Right now it is checking if the primary state is down and then again checking if the primary state is down.  Do you want to fix it here or file a bug for mainline?  It's a pretty small change, so I'd suggest just fixing it.

Reply:
I had addressed it in the last commit. That if condition was redundant from the one below it and only used to break from the loop. Since it is iterating the PSI list which is a small list, I had removed the break.